modules/client/src/internal/client/methods.ts

```diff
@@ -1075,4 +1084,69 @@ export class ClientMethods extends ClientCore implements IClientMethods {
     }
     return version;
   }
+
+  public async isDaoUpdateProposalValid(
+    proposalId: string,
+  ): Promise<DaoUpdateProposalValidity> {
+    if (!isProposalId(proposalId)) {
+      throw new InvalidProposalIdError();
+    }
+    // search for the proposal
+    const name = "IProposal";
+    const query = QueryIProposal;
+    type T = { iproposal: SubgraphIProposal };
+    const { iproposal } = await this.graphql.request<T>({
+      query,
+      params: { id: proposalId.toLowerCase() },
+      name,
+    });
+    if (!iproposal) {
+      throw new ProposalNotFoundError();
+    }
+    const res: DaoUpdateProposalValidity = {
+      version: false,
+      implementation: false,
+      initData: false,
+      actions: false,
+    };
```

Comment:
i would return {isValid: bool, resaons: {....}} but i will let u be the judge of that